modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/FieldsSelectorDropdown.js

```diff
@@ -77,18 +77,33 @@ const FieldsSelectorDropdown = ({fields}) => {
 						<ClayDropDown.Item
 							key={fieldName}
 							onClick={() => {
-								dispatch(
-									persistVisibleFieldNames({
-										appURL,
-										id,
-										portletId,
-										visibleFieldNames: {
-											...selectedFieldNames,
-											[fieldName]: !selectedFieldNames[
-												fieldName
-											],
-										},
-									})
+								const newVisibleFieldNames = {
+									...selectedFieldNames,
+									[fieldName]: !selectedFieldNames[fieldName],
+								};
+
+								Object.keys(newVisibleFieldNames).some(
+									(visibleFieldName) => {
+										if (
+											newVisibleFieldNames[
+												visibleFieldName
+											]
+										) {
+											dispatch(
+												persistVisibleFieldNames({
+													appURL,
+													id,
+													portletId,
+													visibleFieldNames: {
+														...selectedFieldNames,
+														[fieldName]: !selectedFieldNames[
+															fieldName
+														],
+													},
+												})
+											);
+										}
+									}
```

Comment:
I didn't get very deep into understanding the `persistVisibleFieldNames` function but it is a bit strange that `dispatch` will be called multiple times if it has multiple visible elements and I also don't understand why you are using `some` here.

I think to achieve the same goal we could just check if `fieldName` contains the `selectedFieldNames` object and if it is the only item and avoid doing the `dispatch`.

I'm guessing that `selectedFieldNames` just shows the visible items but apparently it doesn't 🤷‍♂️, so you could check that `fieldName` is the only one selected inside the `selectedFieldNames` object before doing the `dispatch`.

Reply:
> I didn't get very deep into understanding the persistVisibleFieldNames function but it is a bit strange that dispatch will be called multiple times if it has multiple visible

No, it will be called just once, when `visibleFieldName` match with newVisibleFieldNames dict. I used Array.some here for avoiding iterating over the whole array of `Object.keys(newVisibleFieldNames)`.

Reply:
We weren't adding a return value to some, so the capability of stopping iterating when finding something was not used 😅 . Thanks for the headsup